### PR TITLE
More natural swipe animations.

### PR DESCRIPTION
**Pull request by dbuezas**
Changes all swipe animation easings to ease-in for the first half and ease-out for the second half

WARNING/SORRY: not tested.

How do you build this?

### CODE REVIEW

**Comment by dbuezas** [comment]
> > How do you build this?
> 
> I have to create a guide, I know, it's in the todo list. Anyway, are you on linux? If so just call `scripts/build.sh`.

You can also add a script section to the package json, that's where the JavaScript folk (e.g me) looks for build scripts :)
`npx rollup --config`

**Comment by dbuezas** [comment]
Btw, I went slightly overboard and removed a lot of repetition in the  code (almost 20 lines less), but the important part are the ease-in and ease-out parameters of the transition.

**Comment by dbuezas** [comment]
Built it yet. It has an issue, don't test/merge yet :)

**Comment by dbuezas** [comment]
I forgot to mention that the last commit works fine, I'm enjoying this component a lot :)

**Comment by zanna-37** [comment]
You accidentally disabled swipe when animation=="none", which is the default. 😂 But don't worry, I fixed it.

I took the liberty of rewording your commits as I'm adhering to conventional commits (yes, I know I need to create a CONTRIBUTING file). I hope it's not a problem.

Anyway, thank you a lot for your contribution! Animations are now way snappier! I'm even thinking about using "swipe" as default.

**Review by zanna-37 — APPROVED** [review]

File: src/swipe-navigation.ts [hunk 1]
```diff
@@ -577,56 +577,51 @@ class swipeManager {
...
+          } else if (Config.animate == "flip") {
+            view.style.transform = "rotatey(90deg)";
+            view.style.opacity = "0.25";
+            setTimeout(function () {
+              view.style.transition = "";
+              tabs[index].dispatchEvent(new MouseEvent("click", { bubbles: false, cancelable: true }));
+            }, duration + 10);
+
+          } else {
+            const exhaustiveCheck: never = Config.animate;
```
Comment (dbuezas) [inline src/swipe-navigation.ts:616]: Cool Trick!

File: src/swipe-navigation.ts [hunk 2]
```diff
@@ -577,56 +577,51 @@ class swipeManager {
...
-        } else if (Config.animate == "fade") {
-          view.style.transitionDuration = "200ms";
-          view.style.transition = "opacity 0.20s";
-          view.style.opacity = "0";
-          setTimeout(function () {
-            tabs[index].dispatchEvent(new MouseEvent("click", { bubbles: false, cancelable: true }));
-            view.style.transitionDuration = "0ms";
+
+        if (Config.animate == "none") {
+          tabs[index].dispatchEvent(new MouseEvent("click", { bubbles: false, cancelable: true }));
```
Comment (dbuezas) [inline src/swipe-navigation.ts:582]: Ups... I think i confused "none" with undefined (i.e. no animation set in the config)
Good catch!

**Comment by dbuezas** [comment]
Happy to hear my changes were worth a merge!
Re commit history rework: I'm rough on git, have been in management for the last 4 or 5 years so catching up. Thanks for mentioning the steps you follow, you inspired me to pay more attention to this (and learn) in my own repos

**Comment by dbuezas** [comment]
And yes, swipe feels nice :) I modified the default to swipe on my HA already.
It was a bit of a surprise that eaxh dashboard needs to be configured individually (i assume technical reasons)

File: src/swipe-navigation.ts (hunk 1 above)
Comment (zanna-37) [reply src/swipe-navigation.ts:616]: Ahahah, thanks ☺️. I love TypeScript's type check, and I try to use it as much as I can.

**Comment by zanna-37** [comment]
> I'm rough on git, have been in management for the last 4 or 5 years so catching up. Thanks for mentioning the steps you follow, you inspired me to pay more attention to this (and learn) in my own repos

Don't worry, and good that you're willing to learn!
I'm very strict when using git 😂 I try to be as clean as possible both with commits and with the graph. I try to create small commits with a single thing in it. In this way, it's easier to navigate the history in the future (and revert mistakes).
For the graph, I tend to follow the principles of [gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).

> It was a bit of a surprise that each dashboard needs to be configured individually (I assume technical reasons)

Yeah, the only way I (well, maykar to be precise) found to have a configuration is to piggyback the dashboard config. But I don't like it too much.
Oh, and be careful, there is a bug that prevent the configuration to update when changing dashboard. It's tracked here #12.

**Comment by zanna-37** [comment]
`v1.6.5` is now in pre-release. Feel free to test it and let me know, so I push it for everyone.
Thanks again!

**Comment by dbuezas** [comment]
v1.6.5 RC Looking fone so fat

**Pull request merged**